src/ethereum/frontier/eth_types.py

```diff
@@ -142,15 +134,70 @@ class Receipt:
 State = Dict[Address, Account]
 
 
+def initialize_account(state: State, address: Address) -> None:
+    """
+    Initialize an address in the state with an empty account.
+    """
+    state[address] = Account(
+        nonce=Uint(0),
+        balance=U256(0),
+        code=bytearray(),
+        storage={},
+    )
+
+
+def get_account(state: State, address: Address) -> Account:
+    """
+    Obtain the account corresponding to the address from the state.
+
+    If the (address, account) pair is not yet present in the state, then a
+    default account would be returned.
+
+    In any case, this function will not modify the state. If you need to
+    modify the state, use `modify_state` function which will automatically
+    take care of initializing the account in the state if the corresponding
+    address doesn't exist in the state.
+    """
+    if address in state:
+        return state[address]
+
+    return Account(
+        nonce=Uint(0),
+        balance=U256(0),
+        code=bytearray(),
+        storage={},
+    )
+
+
 def modify_state(
     state: State, address: Address, f: Callable[[Account], None]
 ) -> None:
     """
     Modify an `Account` in the `State`.
+
+    This will also initialize an account if the passed address is not present
+    in the state yet.
     """
+    if address not in state:
+        initialize_account(state, address)
+
     state[address] = modify(state[address], f)
```

Comment:
I think deleting an account would need another function specified, as it is more explicit calling a `delete_account` function rather than `modify_state` with `f` as a noop.

## Without deleting account from state in `modify_state`
```
def delete_account(state, address):
    def del_acc(address):
        state.pop(address, None)

    modify_state(state, address, del_acc)
```

## With deleting account from state in `modify_state`
```
def delete_account(state, address):
    def noop():
        pass

    modify_state(state, address, pass)
```

Reply:
Calling `modify_state` with a noop wouldn't delete the account from state (unless of course the account was already equal to the empty account.)

As far as I understand it, an empty account and an account not in `state` are exactly equivalent? In other words, does this example hold true:

```python
state = {
    b"\x00\x01...": Account(
        nonce=Uint(0),
        balance=U256(0),
        code=bytearray(),
        storage={},
    ),
}

exists = state[b"\x00\x01..."]
doesnt = state[b"\xFF\xFF..."]

assert exists == doesnt
```

If that example does hold true, I don't think there is any point in storing empty accounts in the state, since missing keys return the empty account anyway.

Reply:
I see your point. I'll include this as well. Thanks.